lib/ansible/galaxy/dependency_resolution/__init__.py

```diff
@@ -21,15 +21,15 @@
 
 
 def build_collection_dependency_resolver(
-        galaxy_apis,  # type: t.Iterable[GalaxyAPI]
-        concrete_artifacts_manager,  # type: ConcreteArtifactsManager
-        preferred_candidates=None,  # type: t.Iterable[Candidate]
-        with_deps=True,  # type: bool
-        with_pre_releases=False,  # type: bool
-        upgrade=False,  # type: bool
-        include_signatures=True,  # type: bool
-        offline=False,  # type: bool
-):  # type: (...) -> CollectionDependencyResolver
+        galaxy_apis: t.Iterable[GalaxyAPI],
+        concrete_artifacts_manager: ConcreteArtifactsManager,
+        preferred_candidates: t.Iterable[Candidate] = None,
```

Comment:
Arguments which accept `None` should include that in their annotation with `| None`.

Reply:
@mattclay isn't that check disabled in MyPy? I know it can infer `| None` in places where `= None` is a default. Do we have a policy of an explicit `| None`?

Reply:
Ack, I wasn't sure if we wanted that. Will do so from now on.